### ISSUE TITLE
Looking into the content

### ISSUE THREAD

**Issue by Shuyib** [BioinfoNet/Data-mining#19]
Hi, 

I've just seen that the repo has been updated to reflect changes till now. I have several questions on the way forward. 

* Are we going to make an application for this? What if someone wants to do this in future? I'm thinking ipywidgets could help.

* Are we going to do use Natural language Processing to look at the main topics being posted on the papers?

What do you think? @kipkurui

**Comment by Shuyib** [comment]
I'll do it anyway. When I get the time.

**Comment by kipkurui** [comment]
Hi @Shuyib that is a wonderful Idea. It would be fun to have a widget that would provide real-time visualisation. Is that what you had in mind. For natural language processing, which kind of question can we prioritise? 
- Are we analysing the abstract only? It can be tricky to get the full text, especially or closed access journals.

**Comment by Shuyib** [comment]
Yes, we'll need to wrap everything in a `while` loop at the querying and returning the results + timing in between. I think it would be better to make one version where the user can just query what they want. I can wrap that with ipywidgets. 

For NLP, I'll use Non-negative matrix factorization which is an unsupervised learning method. We can adjust the `K` that is, the number of clusters to potentially find out the topics in the abstracts only. Unfortunately. Just a by the way, we could use this in the next session [mybinder](https://mybinder.org/)

How does that sound?
